File: android/protectmyphone/app/src/main/java/com/protectmyphone/app/BootReceiver.java
package com.protectmyphone.app;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.VpnService;
import android.os.Build;

public final class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent == null || !Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) return;
        boolean enabled = context.getSharedPreferences("protectmyphone", Context.MODE_PRIVATE)
                .getBoolean("monitor_enabled", false);
        if (!enabled) return;
        if (VpnService.prepare(context) != null) return;

        Intent service = new Intent(context, DomainVpnService.class);
        service.setAction(DomainVpnService.ACTION_START);
        if (Build.VERSION.SDK_INT >= 26) context.startForegroundService(service);
        else context.startService(service);
    }
}

File: missed-call-text-back-app/app/src/main/java/com/ownerslocal/missedcalltextback/service/BootReceiver.kt
package com.ownerslocal.missedcalltextback.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import androidx.core.content.ContextCompat
import com.ownerslocal.missedcalltextback.MissedCallApp

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action != Intent.ACTION_BOOT_COMPLETED) return
        val app = context.applicationContext as MissedCallApp
        if (!app.sessionStore.isSignedIn) return
        ContextCompat.startForegroundService(context, Intent(context, MissedCallMonitorService::class.java))
    }
}
